metsim/methods/mtclim.py

```diff
@@ -106,127 +91,90 @@ def _calc_tfmax(precip, dtr, sm_dtr):
         sc[inds] = (1.32 + 0.096 * df['swe'][inds]) * 1.0e6 / daylength[yday][inds]
         sc = np.maximum(sc, 100.)  # JJH - this is fishy 
-    df['swrad'] = srad1 + srad2 + sc
 
-    potrad = (srad1+srad2+sc) * daylength[yday] / t_final / consts['SEC_PER_DAY'] 
-    tfmax = np.ones(len(sc)) 
-    inds = np.nonzero((potrad > 0.) & (df['swrad'] > 0.) & (daylength[yday] > 0))[0]
-    tfmax[inds] = np.maximum((df['swrad'][inds] / (potrad[inds] * t_tmax[inds])), 1.)
-    df['tfmax'] = tfmax 
+    # Calculation of shortwave is split into 2 components:
+    # 1. Radiation from incident light
+    # 2. Influence of snowpack - optionally set by MTCLIM_SWE_CORR
+    df['swrad'] = potrad[yday] * t_final + sc
 
-    if (options['LW_CLOUD'].upper() == 'CLOUD_DEARDORFF'):
+    # Calculate cloud effect
+    if (cnst.LW_CLOUD.upper() == 'CLOUD_DEARDORFF'):
         df['tskc'] = (1. - df['tfmax'])
     else:
         df['tskc'] = np.sqrt((1. - df['tfmax']) / 0.65)
 
     # Compute PET using SW radiation estimate, and update Tdew, pva **
-    tmink = df['t_min'] + consts['KELVIN']
     pet = calc_pet(df['swrad'], df['t_day'], pa, df['dayl'])
-
-    # calculate ratio (PET/effann_prcp) and correct the dewpoint
+    # Calculate ratio (PET/effann_prcp) and correct the dewpoint
     ratio = pet / parray
-    df['ppratio'] = ratio * 365.25
-    tdewk = tmink * (-0.127+1.121 * (1.003-1.444 * ratio+12.312 *
-                np.power(ratio, 2)-32.766 * np.power(ratio, 3))+0.0006 * dtr)
-    tdew = tdewk - consts['KELVIN']
-
-    pva = svp(tdew)
-    if 'hum' not in df:
-        df['hum'] = pva
-    
-    pvs = svp(df['t_day'])
-    df['vpd'] = np.maximum(pvs-pva, 0.)
-
-
-def calc_longwave(df: pd.DataFrame):
-    """ Calculate longwave """
-    emissivity_calc = {
-            'DEFAULT'    : lambda x : x,
-            'TVA'        : lambda x : 0.74 + 0.0049 * x,
-            'ANDERSON'   : lambda x : 0.68 + 0.036 * np.power(x, 0.5),
-            'BRUTSAERT'  : lambda x : 1.24 * np.power(x/air_temp, 0.14285714),
-            'SATTERLUND' : lambda x : 1.08 * (1 - np.exp(-1 * np.power(x, (air_temp/2016)))),
-            'IDSO'       : lambda x : 0.7 + 5.95e-5 * x * np.exp(1500/air_temp),
-            'PRATA'      : lambda x : (1 - (1 + (46.5 * x/air_temp)) *
-                np.exp(-1*np.power((1.2 + 3 * (46.5*x/air_temp)), 0.5))),
-            }
-    cloud_calc = {
-            'DEFAULT' : lambda x : (1.0 + (0.17 * tskc**2)) * x,
-            'CLOUD_DEARDORFF' : lambda x : tskc * 1.0 + (1-tskc) * x
-            }
-    tskc = df['tskc']
-    air_temp = df['t_day'] + consts['KELVIN'] 
-    emissivity_clear = emissivity_calc[options['LW_TYPE'].upper()](air_temp)
-    emissivity = cloud_calc[options['LW_CLOUD'].upper()](emissivity_clear) 
-    df['lwrad'] = emissivity * consts['STEFAN_B'] * np.power(air_temp, 4)
+    df['pet'] = parray 
+    tmink = df['t_min'] + cnst.KELVIN
+    tdew = tmink*(-0.127 + 1.121*(1.003 - 1.444*ratio + 12.312*np.power(ratio, 2)  
```

Comment:
I think here it's OK to maintain the literal values. Would be good to state where the function comes from.